README.md
# flutter_credit_card_ui

![GitHub](https://img.shields.io/github/license/ingeniousmindslab/flutter_credit_card_ui)

`flutter_credit_card_ui` is a Flutter package that allows you to easily create debit card and credit card user interfaces. With this package, users can also flip the card to view additional details. It provides customizable options for card decoration, such as gradient colors, background images, and various text styles.

## Installation

Add the following dependency to your `pubspec.yaml` file:

```
dependencies:
  flutter_credit_card_ui: ^1.0.0

```

## Usage

Import the package into your Dart file:

```
import 'package:flutter_credit_card_ui/flutter_credit_card_ui.dart';

```

Use the `CreditCardWidget` widget to create a card interface:

```
CreditCardWidget(
  cardDecoration: CardDecoration(
    showBirdImage: true,
  ),
  cvvText: "478",
  gradient: const LinearGradient(
    colors: [
      Color(0xFF3366FF),
      Color(0xFF00CCFF),
    ],
    begin: FractionalOffset(0.0, 0.0),
    end: FractionalOffset(1.0, 0.0),
    stops: [0.0, 1.0],
    tileMode: TileMode.clamp,
  ),
  cardHolder: "Mr. Marcia Fadel",
  cardNumber: "4660 9282 5392 9728",
  cardExpiration: "7/2026",
  cardtype: CardType.rupay,
  color: Colors.red,
),

```

## Showcase

<img src="https://raw.githubusercontent.com/ingeniousmindslab/flutter_credit_card_ui/master/images/showcase.gif" width="250">


## About Us

`flutter_credit_card_ui` is developed and maintained by Ingenious Minds Lab, a software development company specializing in mobile and web applications.

Learn more about us and our projects on our website: [www.ingeniousmindslab.com](https://www.ingeniousmindslab.com/)

## Contributions

Contributions are welcome! If you have any suggestions, bug reports, or feature requests, please create an issue on the [GitHub repository](https://github.com/ingeniousmindslab/flutter_credit_card_ui).

## License

This package is released under the [MIT License](https://chat.openai.com/LICENSE).


CHANGELOG.md
## 0.0.3

* Minor Bug Fix


lib/src/credit_card.dart
import 'package:flip_card/flip_card.dart';
import 'package:flutter/material.dart';
import 'package:flutter_credit_card_ui/src/models/credit_card_model.dart';
import 'package:flutter_credit_card_ui/src/utils/methods.dart';
import 'package:flutter_credit_card_ui/src/utils/utills.dart';

class CreditCardWidget extends StatelessWidget {
  const CreditCardWidget({
    super.key,
    required this.color,
    required this.cardNumber,
    required this.cardHolder,
    required this.cardExpiration,
    required this.cardtype,
    this.cardDecoration,
    this.cvvText,
    this.declarationText,
    this.authoRizedText,
    this.stdText,
    this.gradient,
    this.bgImage,
  });
  final DecorationImage? bgImage;
  final Color color;
  final String cardNumber;
  final String? cvvText;
  final String? stdText;
  final String cardHolder;
  final String? declarationText;
  final String? authoRizedText;
  final Gradient? gradient;
  final String cardExpiration;
  final CardType cardtype;
  final CardDecoration? cardDecoration;
  @override
  Widget build(BuildContext context) {
    return FlipCard(
      front: Center(
        child: SizedBox(
          width: Utils.kCardWidth,
          height: Utils.kCardWidth * Utils.ratio,
          child: Container(
            decoration: BoxDecoration(
                image: bgImage,
                gradient: gradient,
                color: color,
                borderRadius: BorderRadius.circular(14)),
            child: Center(
              child: Container(
                height: Utils.kCardHeight,
                width: Utils.kCardWidth,
                padding: const EdgeInsets.only(
                    left: 16.0, right: 16.0, bottom: 22.0),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: <Widget>[
                    Row(
                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
                      children: [
                        Text(
                          cardname(cardtype),
                          style: cardDecoration?.companyNameStyle ??
                              const TextStyle(
                                color: Colors.white,
                                fontSize: 20,
                                fontWeight: FontWeight.w500,
                                package: "flutter_credit_card_ui",
                                letterSpacing: 1,
                              ),
                        ),
                        Image.asset(
                          getCardLogo(cardtype),
                          height: 60,
                          width: 60,
                          package: "flutter_credit_card_ui",
                        ),
                      ],
                    ),
                    Row(
                      children: [
                        Image.asset(
                          "assets/images/chip.png",
                          height: 50,
                          width: 60,
                          package: "flutter_credit_card_ui",
                        ),
                        const SizedBox(width: 8),
                        Visibility(
                          visible: cardDecoration?.showContactLessIcon ?? true,
                          child: Image.asset(
                            "assets/images/contact_less.png",
                            height: 30,
                            width: 30,
                            package: "flutter_credit_card_ui",
                          ),
                        ),
                      ],
                    ),
                    Text(
                      cardNumber,
                      style: cardDecoration?.cardNumberTextStyle ??
                          const TextStyle(
                            color: Colors.white,
                            fontSize: 21,
                            letterSpacing: 4,
                            fontWeight: FontWeight.w500,
                          ),
                    ),
                    Row(
                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
                      children: <Widget>[
                        buildCardHolderDetailsBlock(
                          textStyle: cardDecoration?.cardHolderLabelStyle,
                          detailstextStyle: cardDecoration?.cardNumberTextStyle,
                          label: 'CARDHOLDER',
                          value: cardHolder,
                        ),
                        buildvalidDetailsBlock(
                            detailstextStyle:
                                cardDecoration?.validTillDetailsStyle,
                            labeltextStyle: cardDecoration?.validTillLabelStyle,
                            label: 'VALID THRU',
                            value: cardExpiration),
                      ],
                    ),
                  ],
                ),
              ),
            ),
          ),
        ),
      ),
      back: Container(
        margin: const EdgeInsets.symmetric(vertical: 16),
        child: Center(
          child: Container(
            width: Utils.kCardWidth,
            height: Utils.kCardWidth * Utils.ratio,
            padding: const EdgeInsets.symmetric(vertical: 5),
            decoration: BoxDecoration(
                image: bgImage,
                gradient: gradient,
                color: color,
                borderRadius: BorderRadius.circular(18)),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                const SizedBox(height: 8),
                Container(
                  height: 40,
                  width: Utils.kCardWidth,
                  color: Colors.black,
                ),
                const SizedBox(height: 15),
                Row(
                  children: [
                    SizedBox(
                        width: cardDecoration?.showBirdImage == false ? 50 : 5),
                    Visibility(
                      visible: cardDecoration?.showBirdImage ?? true,
                      child: Image.asset(
                        "assets/images/bird.png",
                        height: 50,
                        width: 50,
                        package: "flutter_credit_card_ui",
                      ),
                    ),
                    Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Text(
                          stdText ?? "Prifix Local STD call",
                          style: cardDecoration?.stdTextStyle ??
                              const TextStyle(color: Colors.white, fontSize: 8),
                        ),
                        const SizedBox(height: 3),
                        Row(
                          children: [
                            Container(
                              height: 40,
                              width: Utils.kCardWidth * .5,
                              color: Colors.grey[600],
                            ),
                            Container(
                              height: 25,
                              width: Utils.kCardWidth * .1,
                              color: Colors.white,
                              child: Center(
                                  child: Text(
                                cvvText ?? "000",
                                style: cardDecoration?.cvvTextStyle ??
                                    const TextStyle(color: Colors.black),
                              )),
                            ),
                          ],
                        ),
                        const SizedBox(height: 3),
                        SizedBox(
                          width: 300,
                          child: Text(
                            authoRizedText ?? "AUTHORISED SIGNATURE",
                            maxLines: 2,
                            style: cardDecoration?.authoRizedTextStyle ??
                                const TextStyle(
                                    color: Colors.white, fontSize: 10),
                          ),
                        ),
                      ],
                    ),
                  ],
                ),
                Padding(
                  padding: const EdgeInsets.only(left: 10, top: 12, right: 10),
                  child: Text(
                    maxLines: 6,
                    declarationText ??
                        "Never share your credit card Number, Expiry Date, CVV, PIN & OTP With anyone.To Generate manage pin, download Card Mobile APp ir visit \n\nNever share your credit card PINs, interest banking passwords, or mobile banking passwords with anyone, be it a friend or a family member, While not illegal to do so, it is certainly foolish to share your credit or debit card information via email or text",
                    style: cardDecoration?.declarationTextStyle ??
                        const TextStyle(
                            color: Colors.white,
                            fontSize: 8,
                            fontWeight: FontWeight.w400),
                  ),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}

Column buildCardHolderDetailsBlock(
    {required String label,
    required String value,
    TextStyle? textStyle,
    TextStyle? detailstextStyle}) {
  return Column(
    crossAxisAlignment: CrossAxisAlignment.start,
    children: <Widget>[
      Text(
        label,
        style: textStyle ??
            const TextStyle(
                color: Colors.grey, fontSize: 10, fontWeight: FontWeight.bold),
      ),
      Text(
        value,
        style: detailstextStyle ??
            const TextStyle(
              color: Colors.white,
              fontSize: 15,
              fontWeight: FontWeight.w600,
              letterSpacing: .5,
            ),
      )
    ],
  );
}

Column buildvalidDetailsBlock(
    {required String label,
    required String value,
    TextStyle? labeltextStyle,
    TextStyle? detailstextStyle}) {
  return Column(
    crossAxisAlignment: CrossAxisAlignment.start,
    children: <Widget>[
      Text(
        label,
        style: labeltextStyle ??
            const TextStyle(
                color: Colors.grey, fontSize: 10, fontWeight: FontWeight.bold),
      ),
      Text(
        value,
        style: detailstextStyle ??
            const TextStyle(
              color: Colors.white,
              fontSize: 15,
              fontWeight: FontWeight.w600,
              letterSpacing: .5,
            ),
      )
    ],
  );
}


lib/src/utils/methods.dart
import 'package:flutter_credit_card_ui/src/utils/utills.dart';

String cardname(CardType type) {
  switch (type) {
    case CardType.visa:
      return "Visa";
    case CardType.americanExpress:
      return "American Express";
    case CardType.discover:
      return "Discover";
    case CardType.masterCard:
      return "Mastercard";
    case CardType.rupay:
      return "RuPay";

    default:
      return "UnKnown";
  }
}

String getCardLogo(CardType type) {
  switch (type) {
    case CardType.visa:
      return "assets/logos/visa.png";
    case CardType.americanExpress:
      return "assets/logos/amex.png";
    case CardType.discover:
      return "assets/logos/discover.png";
    case CardType.masterCard:
      return "assets/logos/mastercard.png";
    case CardType.rupay:
      return "assets/logos/rupay.png";

    default:
      return "UnKnown";
  }
}


lib/src/models/credit_card_model.dart
import 'package:flutter/material.dart';

class CreditCardData {}

class CardDecoration {
  final bool? showBirdImage;
  final bool? showContactLessIcon;
  final TextStyle? companyNameStyle;
  final TextStyle? cvvTextStyle;
  final TextStyle? cardHolderLabelStyle;
  final TextStyle? validTillLabelStyle;
  final TextStyle? authoRizedTextStyle;
  final TextStyle? cardHolderDetailsStyle;
  final TextStyle? validTillDetailsStyle;
  final TextStyle? cardNumberTextStyle;
  final TextStyle? stdTextStyle;
  final TextStyle? declarationTextStyle;

  CardDecoration({
    this.authoRizedTextStyle,
    this.declarationTextStyle,
    this.cvvTextStyle,
    this.showBirdImage,
    this.stdTextStyle,
    this.cardNumberTextStyle,
    this.cardHolderDetailsStyle,
    this.validTillDetailsStyle,
    this.showContactLessIcon,
    this.cardHolderLabelStyle,
    this.validTillLabelStyle,
    this.companyNameStyle,
  });
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:flutter_credit_card_ui/flutter_credit_card_ui.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      debugShowCheckedModeBanner: false,
      home: CreditCardUIWidget(),
    );
  }
}

class CreditCardUIWidget extends StatelessWidget {
  const CreditCardUIWidget({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        crossAxisAlignment: CrossAxisAlignment.center,
        children: [
          CreditCardWidget(
            cardDecoration: CardDecoration(showBirdImage: true),
            cvvText: "478",
            gradient: const LinearGradient(
                colors: [
                  Color(0xFF3366FF),
                  Color(0xFF00CCFF),
                ],
                begin: FractionalOffset(0.0, 0.0),
                end: FractionalOffset(1.0, 0.0),
                stops: [0.0, 1.0],
                tileMode: TileMode.clamp),
            cardHolder: "Mr. Marcia Fadel",
            cardNumber: "4660 9282 5392 9728",
            cardExpiration: "7/2026",
            cardtype: CardType.rupay,
            color: Colors.red,
          ),
        ],
      ),
    );
  }
}
